src/hooks/core/getstream/usePostNotificationListenerHook.ts

```diff
@@ -7,34 +7,48 @@ import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
 import useProfileHook from '../profile/useProfileHook';
 import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
 
-const usePostNotificationListenerHook = (onPostNotifReceived) => {
+const usePostNotificationListenerHook = (onPostNotifReceived, isAnonymous = true) => {
   const {localDb} = useLocalDatabaseHook() as UseLocalDatabaseHook;
   const feedSubscriptionRef = React.useRef<StreamFeed<UR, UR, UR, UR, UR, UR> | undefined>(
     undefined
   );
-  const {anonProfileId} = useProfileHook();
+
+  const {anonProfileId, signedProfileId} = useProfileHook();
 
   const initFeedSubscription = async () => {
-    const token: string = TokenStorage.get(ITokenEnum.anonymousToken);
-    const client = clientStream(token);
+    const anonymousToken: string = TokenStorage.get(ITokenEnum.anonymousToken);
+    const signedToken: string = TokenStorage.get(ITokenEnum.token);
+
+    const tokenToSubscribe = isAnonymous ? anonymousToken : signedToken;
+    const idToSubscribe = isAnonymous ? anonProfileId : signedProfileId;
+
+    const client = clientStream(isAnonymous ? anonymousToken : signedToken);
     try {
-      const notifFeed = client?.feed('notification', anonProfileId, token);
+      const notifFeed = client?.feed('notification', idToSubscribe, tokenToSubscribe);
       notifFeed?.subscribe((data) => {
         if (!data || !onPostNotifReceived || typeof onPostNotifReceived !== 'function') return;
         onPostNotifReceived(data);
       });
 
       feedSubscriptionRef.current = notifFeed;
     } catch (e) {
-      console.log('anonProfileId', anonProfileId);
-      console.log('token', token);
+      console.log('profileId', isAnonymous ? anonProfileId : signedProfileId);
+      console.log('token', isAnonymous ? anonymousToken : signedToken);
       console.log('initFeedSubscription error');
       console.log(e);
     }
   };
 
   React.useEffect(() => {
-    if (anonProfileId) initFeedSubscription();
+    if (anonProfileId && isAnonymous) initFeedSubscription();
+
+    return () => {
+      feedSubscriptionRef.current?.unsubscribe();
+    };
+  }, [localDb, anonProfileId]);
+
+  React.useEffect(() => {
+    if (anonProfileId && !isAnonymous) initFeedSubscription();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `usePostNotificationListenerHook` function has been updated to handle both anonymous and signed users. However, there are a couple of issues that need to be addressed:

1. The `initFeedSubscription` function is called in two separate `useEffect` hooks based on the value of `isAnonymous`. This could lead to unnecessary calls if `anonProfileId` changes while the component is mounted. It would be better to have a single `useEffect` hook that handles both cases.

2. Error handling in the `catch` block of the `initFeedSubscription` function is not sufficient. Simply logging the error to the console may not be enough for production code. Consider using a more robust error handling strategy, such as displaying an error message to the user or retrying the operation after some delay.

Here's a suggested fix:

```diff
-  React.useEffect(() => {
-    if (anonProfileId && isAnonymous) initFeedSubscription();
-
-    return () => {
-      feedSubscriptionRef.current?.unsubscribe();
-    };
-  }, [localDb, anonProfileId]);
-
-  React.useEffect(() => {
-    if (anonProfileId && !isAnonymous) initFeedSubscription();
-
-    return () => {
-      feedSubscriptionRef.current?.unsubscribe();
-    };
-  }, [localDb, anonProfileId]);

+  React.useEffect(() => {
+    if (anonProfileId) initFeedSubscription();
+
+    return () => {
+      feedSubscriptionRef.current?.unsubscribe();
+    };
+  }, [localDb, anonProfileId, isAnonymous]);
```

For the error handling part, you might want to consider adding a state variable to store the error and display it to the user, or use a library like `react-query` which provides built-in mechanisms for error handling and retries.


<!-- This is an auto-generated comment by OSS CodeRabbit -->